testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/ProcessInstanceAssertj.java

```diff
@@ -92,19 +119,97 @@ private void hasProcessInstanceInState(
     }
   }
 
+  private void hasElementsInState(
```

Comment:
❌ Some fun edge-case here is when there is multiple elements with the same name in the process, or if there is a loop in the process. When we pass the same element twice, I assume the endpoint returns 2 flownode instances. What will happen if the first flownode is completed and the second is active? My expectation is that the assertion will always fail 😄 

If this case is too complex for this PR just ignore it, but we should be aware of the limitation for now.

Reply:
> What will happen if the first flownode is completed and the second is active? My expectation is that the assertion will always fail

Nope. It would pass the since at lease one flow node is in the expected state. :sweat_smile: 

I would like to ignore this case for now. We could solve this case by introducing another method: `hasCompletedElements("A", 2)`, similar to ZPA.